pkg/api/impl/coinmarketcap/coinmarketcap.go

```diff
@@ -77,8 +77,7 @@ func (s *Service) getPaginatedCoinData(convert string, offset int) ([]apitypes.C
 		}
 
 		ret = append(ret, apitypes.Coin{
-			// TODO: Fix ID
-			ID:               util.FormatID(v.Name),
+			ID:               util.FormatID(fmt.Sprint(v.ID)),
```

Comment:
What is the reason for Sprintf over strconv.Itoa()?

Reply:
the ID is now float64 … no other reason :(

Reply:
You're passing the **value** as the **format** to sprintf?   Perhaps strconv.ParseFloat() then?

Reply:
Nah, I use`Sprint` not `Sprintf`, that works fine for CMC `id` because it's integer. Example: https://goplay.tools/snippet/-KaLEhPz8BQ

Reply:
Ah!